local_planner/src/nodes/common.cpp

```diff
@@ -207,16 +193,51 @@ double getAngularVelocity(double desired_yaw, double curr_yaw) {
   return 0.5 * vel;
 }
 
+Eigen::Vector3f toEigen(const geometry_msgs::Point& p) {
+  Eigen::Vector3f ev3(p.x, p.y, p.z);
+  return ev3;
+}
+Eigen::Vector3f toEigen(const pcl::PointXYZ& p) {
+  Eigen::Vector3f ev3(p.x, p.y, p.z);
+  return ev3;
+}
+Eigen::Vector3f toEigen(const geometry_msgs::Vector3& p) {
+  Eigen::Vector3f ev3(p.x, p.y, p.z);
+  return ev3;
+}
+
+geometry_msgs::Point toPoint(const Eigen::Vector3f& ev3) {
+  geometry_msgs::Point gmp;
+  gmp.x = ev3.x();
+  gmp.y = ev3.y();
+  gmp.z = ev3.z();
+  return gmp;
+}
+pcl::PointXYZ toXYZ(const Eigen::Vector3f& ev3) {
+  pcl::PointXYZ gmp;
+  gmp.x = ev3.x();
+  gmp.y = ev3.y();
+  gmp.z = ev3.z();
+  return gmp;
+}
+geometry_msgs::Vector3 toVector3(const Eigen::Vector3f& ev3) {
+  geometry_msgs::Vector3 gmp;
+  gmp.x = ev3.x();
+  gmp.y = ev3.y();
+  gmp.z = ev3.z();
+  return gmp;
+}
+
 Eigen::Vector3f convert(const geometry_msgs::Point& p) {
   Eigen::Vector3f ev3(p.x, p.y, p.z);
   return ev3;
 }
 
-geometry_msgs::Point convert(const Eigen::Vector3f& p) {
+geometry_msgs::Point convert(const Eigen::Vector3f& ev3) {
```

Comment:
Also, I believe this is legacy since the addition of `toPoint()` ?

Reply:
Good catch!